src/poetry/repositories/link_sources/html.py

```diff
@@ -25,8 +29,13 @@ def __init__(self, url: str, content: str) -> None:
 
         self._parsed = html5lib.parse(content, namespaceHTMLElements=False)
 
-    @property
-    def links(self) -> Iterator[Link]:
+    @cached_property
+    def _link_cache(
+        self,
+    ) -> defaultdict[NormalizedName, defaultdict[Version, list[Link]]]:
```

Comment:
I find this really messy -- would you object to making this a named type (`LinkCache` maybe)?